src/components/NewCommentForm/NewCommentForm.tsx

```diff
@@ -1,39 +1,97 @@
-import React from 'react';
+import React, { useState, ChangeEvent } from 'react';
+import { usePostsContext } from '../../customHooks/usePostsContext';
 import './NewCommentForm.scss';
 
-export const NewCommentForm: React.FC = () => (
-  <form className="NewCommentForm">
-    <div className="form-field">
-      <input
-        type="text"
-        name="name"
-        placeholder="Your name"
-        className="NewCommentForm__input"
-      />
-    </div>
-
-    <div className="form-field">
-      <input
-        type="text"
-        name="email"
-        placeholder="Your email"
-        className="NewCommentForm__input"
-      />
-    </div>
-
-    <div className="form-field">
-      <textarea
-        name="body"
-        placeholder="Type comment here"
-        className="NewCommentForm__input"
-      />
-    </div>
-
-    <button
-      type="submit"
-      className="NewCommentForm__submit-button button"
-    >
-      Add a comment
-    </button>
-  </form>
-);
+type ChangeEventHTML = HTMLInputElement | HTMLTextAreaElement;
+
+export const NewCommentForm: React.FC = React.memo(() => {
+  const { comments, setComments, selectedPostId } = usePostsContext();
+  const [userName, setUserName] = useState('');
+  const [userEmail, setUserEmail] = useState('');
+  const [userComment, setUserComment] = useState('');
+
+  const resetForm = () => {
+    setUserName('');
+    setUserEmail('');
+    setUserComment('');
+  };
+
+  const handleChange = (event: ChangeEvent<ChangeEventHTML>) => {
+    const { name, value } = event.target;
+
+    switch (name) {
+      case 'userName':
+        setUserName(value);
+        break;
+      case 'userEmail':
+        setUserEmail(value);
+        break;
+      case 'userComment':
+        setUserComment(value);
+        break;
+      default:
+        break;
+    }
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    const newComment = {
+      id: new Date().valueOf(),
+      postId: selectedPostId,
+      body: userComment,
+      name: userName,
+      email: userEmail,
+    };
+
+    setComments([...comments, newComment]);
```

Comment:
Yes. I undestand. It is 500 error if you use postComment. But it's not our fault. Ideally, we must save the comment into the database.

Reply:
agree